Objects/stringlib/fastsearch.h

```diff
@@ -773,6 +789,9 @@ FASTSEARCH(const STRINGLIB_CHAR* s, Py_ssize_t n,
         else if (mode == FAST_RSEARCH)
             return STRINGLIB(rfind_char)(s, n, p[0]);
         else {
+            if (maxcount == PY_SSIZE_T_MAX) {
```

Comment:
```suggestion
            if (maxcount >= n) {
```

Reply:
Maxcount is only used in the replace function, it is very unlikely that this condition will ever be triggered.

Reply:
OK, but there's no reason to check for `PY_SSIZE_T_MAX` specifically when this works as well.

Reply:
Yes you are correct. However, this function needs some refactoring, as this maxcount provision is only there for replace. Replace for single characters is special.cased elsewhere, so maxcount is actually always Pyssize_t_max I think. I want to revisit this at a later point.